packages/tizen_audio_manager/tizen/src/tizen_audio_manager_plugin.cc

```diff
@@ -89,180 +54,123 @@ class AudioManagerTizenPlugin : public flutter::Plugin {
+    try {
+      audio_manager_.AddVolumeChangedCallback(
+          [events = events_.get()](std::string type_str, int32_t volume) {
+            flutter::EncodableMap msg;
+            msg[flutter::EncodableValue("type")] =
+                flutter::EncodableValue(type_str);
+            msg[flutter::EncodableValue("level")] =
+                flutter::EncodableValue(static_cast<int32_t>(volume));
+            events->Success(flutter::EncodableValue(msg));
+          });
+    } catch (const AudioManagerError &error) {
+      result_->Error(error.code(), error.message());
     }
   }
 
-  void UnregisterObserver() {
-    LOG_INFO("UnregisterObserver");
+  void UnregisterObserver() { audio_manager_.RemoveVolumeChangedCallback(); }
 
-    if (this->volume_changed_cb_is_registered_) {
-      sound_manager_remove_volume_changed_cb(this->volume_changed_cb_id_);
-      this->events_ = nullptr;
-      this->volume_changed_cb_is_registered_ = false;
+  void GetCurrentPlaybackType() {
+    try {
+      std::string type = audio_manager_.getCurrentPlaybackType();
+      result_->Success(flutter::EncodableValue(type));
+    } catch (const AudioManagerError &error) {
+      result_->Error(error.code(), error.message());
     }
   }
 
-  void GetCurrentPlaybackType(MethodResultPtr result) {
-    sound_type_e type;
-    const int ret = sound_manager_get_current_sound_type(&type);
-
-    if (ret == SOUND_MANAGER_ERROR_NONE) {
-      std::string type_str;
-
-      if (ConvertSoundTypeToString(type, type_str)) {
-        result->Success(flutter::EncodableValue(type_str));
-      } else {
-        LOG_ERROR("Failed to convert sound type to string.");
-        result->Error("UnknownSoundType",
-                      "Failed to convert sound type to string");
-      }
-    } else if (ret == SOUND_MANAGER_ERROR_NO_PLAYING_SOUND) {
-      result->Success(flutter::EncodableValue(std::string("none")));
-    } else {
-      const char *message = get_error_message(ret);
-      LOG_ERROR("Failed to get current playback type: %s", message);
-      result->Error(std::to_string(ret), "Failed to get current playback type.",
-                    flutter::EncodableValue(std::string(message)));
-    }
-  }
-
-  void GetMaxLevel(const flutter::EncodableValue &arguments,
-                   MethodResultPtr result) {
-    std::string type_str;
-    sound_type_e type;
-    int max;
-
-    if (!ExtractValueFromMap(arguments, "type", type_str)) {
-      result->Error("InvalidArguments", "Please check type");
-      return;
-    }
-
-    if (!ConvertStringToSoundType(type_str, type)) {
-      LOG_ERROR("Failed to convert string to sound type.");
-      result->Error("UnknownSoundType",
-                    "Failed to convert string to sound type.");
+  void GetMaxLevel(const flutter::EncodableMap *arguments) {
+    std::string type;
+    if (!GetValueFromEncodableMap(arguments, "type", type)) {
+      result_->Error("InvalidArguments", "Please check type");
```

Comment:
nit: Not mandatory but I usually follow this convention.
```suggestion
      result_->Error("Invalid arguments", "No type provided.");
```

Reply:
done